stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/imagepreview/ImagePreviewActivity.kt

```diff
@@ -141,35 +265,245 @@ public class ImagePreviewActivity : AppCompatActivity() {
+            )
+
+            Spacer(modifier = Modifier.width(8.dp))
+
             Text(
-                text = SimpleDateFormat.getTimeInstance().format(message.createdAt ?: Date()),
-                style = ChatTheme.typography.footnote,
-                color = ChatTheme.colors.textLowEmphasis
-            ) // TODO format properly
+                text = imagePreviewOption.title,
+                color = imagePreviewOption.titleColor,
+                style = ChatTheme.typography.bodyBold,
+                fontSize = 12.sp
+            )
+        }
+    }
+
+    /**
+     * Consumes the action user selected to perform for the current image.
+     *
+     * @param imagePreviewAction The action the user selected.
+     * @param currentPage The index of the current image.
+     */
+    private fun handleImageAction(
+        imagePreviewAction: ImagePreviewAction,
+        currentPage: Int,
+    ) {
+        val message = imagePreviewAction.message
+
+        when (imagePreviewAction) {
+            is ShowInChat -> {
+                handleResult(
+                    ImagePreviewResult(
+                        messageId = message.id,
+                        resultType = ImagePreviewResultType.SHOW_IN_CHAT
+                    )
+                )
+            }
+            is Reply -> {
+                handleResult(ImagePreviewResult(messageId = message.id, resultType = ImagePreviewResultType.QUOTE))
+            }
+            is Delete -> imagePreviewViewModel.deleteCurrentImage(message.attachments[currentPage])
+            is SaveImage -> {
+                ChatDomain
+                    .instance()
+                    .downloadAttachment(message.attachments[currentPage])
+                    .enqueue()
+            }
+        }
+    }
+
+    /**
+     * Prepares and sets the result of this Activity and propagates it back to the user.
+     *
+     * @param result The chosen action result.
+     */
+    private fun handleResult(result: ImagePreviewResult) {
+        val data = Intent().apply {
+            putExtra(KEY_IMAGE_PREVIEW_RESULT, result)
         }
+        setResult(RESULT_OK, data)
+        finish()
     }
 
+    /**
+     * Renders a horizontal pager that shows images and allows the user to swipe them and zoom in.
+     *
+     * @param pagerState The state of the pager, that represents the page count and the current page.
+     * @param attachments The attachments to show.
+     */
     @Composable
-    private fun ImagePreviewContent(pagerState: PagerState, message: Message) {
+    private fun ImagePreviewContent(pagerState: PagerState, attachments: List<Attachment>) {
         HorizontalPager(modifier = Modifier.background(ChatTheme.colors.appBackground), state = pagerState) { page ->
 
-            if (message.attachments.isNotEmpty()) {
-                val painter = rememberImagePainter(data = message.attachments[page].imagePreviewUrl)
+            if (attachments.isNotEmpty()) {
```

Comment:
An early return could simplify this method `if (attachments.isEmpty()) return`

Reply:
I think actually both an early return and a `finish()` call would be nicer, because the user can essentially remove all the images.

Reply:
Will add, also added a fix/improvement when trying to delete the last image in the message.